src/checkers/inference/solver/backend/AbstractFormatTranslator.java

```diff
@@ -24,15 +24,18 @@
 import checkers.inference.util.ConstraintVerifier;
 
 /**
- * Created by mier on 08/11/17.
+ * Abstract implementation of FormatTranslator that all concrete FormatTranslator should extend from.
+ * It has the necessary lattice that will be used and already setup the abstract structure and provided
+ * default implementations so that you only need to override minimum methods your FormatTranslator is
+ * interested.
  */
 public abstract class AbstractFormatTranslator<SlotEncodingT, ConstraintEncodingT, SlotSolutionT>
         implements FormatTranslator<SlotEncodingT, ConstraintEncodingT, SlotSolutionT>{
 
-    // lattice is protected because every subclass of AbstractFormatTranslator uses it.
+    /* lattice is protected because every subclass of AbstractFormatTranslator uses it.*/
     protected final Lattice lattice;
-    // verifier is private because subclasses doesn't use it. It only needs to be passed
-    // as an argument to encoder factory methods
+    /* verifier is private because subclasses doesn't use it. It only needs to be passed
```

Comment:
Instead of explaining the visibility modifiers, document what this field is (this would be more helpful for others read your code, as they want to know what this field is about).

E.g.
```java
/**
* A verifier that verify if constraints between constant slots satisfiable.
*
* This verifier helps in optimizing encoding of constraints between constant slots.
* E.g. Instead of directly encode a constant constraint, we encode an empty value if
* it passed verification by this verifier, otherwise encode a contradictory value for it.
*/
private final ConstraintVerifier verifier;
```

Reply:
Personally, I prefer having no documentation on fields at all. For me, you can jump to class Lattice and see the documentation there. If we document fields lattice and verifier, it looks duplicate for me. Do you think having documentation just because of it is a field of another class would help? If so, I can change it to rigorous javadoc.